### ISSUE TITLE
Pyinfra tries to collect facts on all hosts

### ISSUE THREAD

**Issue by giuppep** [Fizzadar/pyinfra#619]
## Describe the bug
I'm trying to use the [`postgresql.database`](https://docs.pyinfra.com/en/1.x/operations/postgresql.html#postgresql-database) operation to create a database if it's not present on deployment. This should run only on the database server. The other servers do not have `postgresql` installed/running.

```python
if "db" in host.groups:
    postgresql.database("my_db", present=True, sudo_user="postgres")
```
On deployment, `pyinfra` runs the [`postgresql.PostgresqlDatabases`](https://docs.pyinfra.com/en/1.x/facts/postgresql.html#postgresql-postgresqldatabases) fact on all servers. This fails on the servers that don't have postgres

```
[no-db-server.my_domain.com] sudo: unknown user: postgres
[no-db-server.my_domain.com] sudo: unable to initialize policy plugin
    [no-db-server.my_domain.com] Error: could not load fact: postgresql_databases postgresql_user=None, postgresql_password=None, postgresql_host=None, postgresql_port=None
```

This causes the script to stop on those servers and not execute the subsequent operations.

## To Reproduce
`inventory.py`
```python
hosts = ["db_server", "web_server"]
db = ["db_server"]
web = ["web_server"]
```
`task.py`
```python
from pyinfra import host
from pyinfra.operations import postgresql

if "db" in host.groups:
    postgresql.database("my_db", present=True, sudo_user="postgres")
```

`pyinfra inventory.py task.py`

## Expected behavior
Pyinfra should not try to collect facts/should ignore errors on hosts that won't be updated

## Meta
+ Include output of `pyinfra --support`.
```
    System: Darwin
      Platform: Darwin-20.4.0-x86_64-i386-64bit
      Release: 20.4.0
      Machine: x86_64
    pyinfra: v1.4.7
    Executable: /Users/giuppep/.pyenv/versions/3.7.5/bin/pyinfra
    Python: 3.7.5 (CPython, Clang 10.0.1 (clang-1001.0.46.4))
```
+ How was pyinfra installed (source/pip)? `pip`
+ Include pyinfra-debug.log (if one was created)
+ Consider including output with `-vv` and `--debug`.

**Comment by Fizzadar** [comment]
Hi @giuppep! So part of pyinfra's speed comes from the way it runs each fact on each host in parallel - this should not cause a failure though! In fact, there's [specific handling of this situation here](https://github.com/Fizzadar/pyinfra/blob/current/pyinfra/api/facts.py#L292-L305) - but in this case the user isn't going to be added (as far as pyinfra knows).

I also think this issue could crop up in a thousand other ways, the `will_add_user` call is "best effort" and realistically there are many missed users that may be created by installing certain packages/etc.

So I think the correct handling here is that pyinfra should never error for user errors when executing facts.

**Comment by giuppep** [comment]
Hey, thanks for the reply! I agree, not erroring would solve the problem without sacrificing performance. Let me know if you need any more info/I can help in any way :)

**Comment by Fizzadar** [comment]
Hi @giuppep this has now been removed in https://github.com/Fizzadar/pyinfra/commit/f9490d347fb1e5132e1454053e85089d7c933917 and released in v1.4.8!

**Comment by giuppep** [comment]
Just tested and now it works as expected. Thanks a lot for the very quick fix :)
